mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -780,12 +781,23 @@ public void updateAnnotations(MailboxPath mailboxPath, MailboxSession session, L
         for (MailboxAnnotation annotation : mailboxAnnotations) {
             if (annotation.isNil()) {
                 annotationMapper.deleteAnnotation(mailboxId, annotation.getKey());
-            } else {
+            } else if (isValid(mailboxId, annotation, annotationMapper)) {
```

Comment:
You should allow annotations update if it decrease annotation size, even over the limit
Otherwise, if some obscur conflict occurs, one can not get under the quota, in some scenario.